src/test/java/org/mockitousage/annotation/SpyAnnotationTest.java

```diff
@@ -173,18 +192,86 @@ public void should_report_when_encosing_instance_is_needed() throws Exception {
         }
     }
 
-    static class NestedClassWithoutDefinedConstructor { }
+    @Test
+    public void should_report_private_inner_not_supported() throws Exception {
+        try {
+            MockitoAnnotations.initMocks(new WithInnerPrivate());
+            fail();
+        } catch (MockitoException e) {
+            // Currently fails at instantiation time, because the mock subclass don't have the
+            // 1-arg constructor expected for the outerclass.
+            // org.mockito.internal.creation.instance.ConstructorInstantiator.withParams()
+            assertThat(e).hasMessageContaining("Unable to initialize @Spy annotated field 'spy_field'")
```

Comment:
The problem is that `InnerPrivate` only declares a `private` constructor. Even if private classes are compiled to be package-private (there are no private classes in Java) what allows subclassing them, you cannot call any constructor of this class from a subclass. Therefore, Byte Buddy does not add a constructor, it is not possible to legally do so.